homeassistant/components/mvglive/sensor.py

```diff
@@ -107,17 +93,23 @@ def __init__(
         self._station = station
         self._name = name
         self.data = MVGLiveData(
-            station, destinations, directions, lines, products, timeoffset, number
+            station, destinations, lines, products, timeoffset, number
         )
         self._state = None
-        self._icon = ICONS["-"]
+        self._icon = NONE_ICON
+
+    @property
+    def entity_id(self):
+        """Return the entity_id of the sensor."""
+        stripped_station = re.sub(r"\W+", "_", self._station).lower()
+        return ENTITY_ID_FORMAT.format(stripped_station)
 
     @property
     def name(self):
         """Return the name of the sensor."""
         if self._name:
             return self._name
-        return self._station
+        return "MVG station sensor: " + self._station
```

Comment:
Not relevant for this PR to change this.
Make it in a follow-up PR and use the short hand attributes instead.